sophus: fails in do_compile on x86-64 architecture

`bitbake sophus` fails during compiling with:
```
[...]
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/core2-64-oe-linux/sophus/0.9.1-r0/recipe-sysroot/usr/include/eigen3/Eigen/src/Core/Matrix.h:311:7:   required from 'Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols>::Matrix(const Eigen::ReturnByValue<OtherDerived>&) [with OtherDerived = Eigen::MatrixExponentialReturnValue<Eigen::Matrix<float, 4, 4> >; _Scalar = float; int _Rows = 4; int _Cols = 4; int _Options = 0; int _MaxRows = 4; int _MaxCols = 4]'
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/core2-64-oe-linux/sophus/0.9.1-r0/sophus-0.9.1/sophus/tests.hpp:96:64:   required from 'bool Sophus::Tests<LieGroup>::expMapTest() [with LieGroup = Sophus::Sim3Group<float>]'
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/core2-64-oe-linux/sophus/0.9.1-r0/sophus-0.9.1/sophus/tests.hpp:211:24:   required from 'void Sophus::Tests<LieGroup>::runAllTests() [with LieGroup = Sophus::Sim3Group<float>]'
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/core2-64-oe-linux/sophus/0.9.1-r0/sophus-0.9.1/sophus/test_sim3.cpp:97:3:   required from 'void tests() [with Scalar = float]'
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/core2-64-oe-linux/sophus/0.9.1-r0/sophus-0.9.1/sophus/test_sim3.cpp:107:16:   required from here
| /home/lukas/work/repositories/openembedded.org/openembedded-core/build/tmp-glibc/work/core2-64-oe-linux/sophus/0.9.1-r0/recipe-sysroot/usr/include/eigen3/Eigen/src/Core/DenseCoeffsBase.h:56:30: error: ignoring attributes on template argument 'Eigen::internal::packet_traits<float>::type {aka __vector(4) float}' [-Werror=ignored-attributes]
| ERROR: oe_runmake failed
```

The first build configuration I noticed the issue was:
```
Build Configuration:
BB_VERSION        = "1.33.4"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "ubuntu-14.04"
TARGET_SYS        = "x86_64-oe-linux"
MACHINE           = "qemux86-64"
DISTRO            = "nodistro"
DISTRO_VERSION    = "nodistro.0"
TUNE_FEATURES     = "m64 core2"
TARGET_FPU        = ""
meta              = "master:e584be78f92ee6f08f570c239698d56ac78d05f9"
meta-oe           
meta-multimedia   
meta-python       = "master:2c21aa8633e4e3538099bcc56d76ab0a594411b4"
meta-ros          = "master:f162d103a3b0fa97133fd15dd56eca34fcdf264e"
```
The issue however occurs since sophus' initial commit in meta-ros at 7b08f250691e09c2ec6ac8b7d45713ee14a86813.

--- a/recipes-extended/sophus/sophus_0.9.1.bb
+++ b/recipes-extended/sophus/sophus_0.9.1.bb
@@ -14,4 +14,4 @@ S = "${WORKDIR}/sophus-${PV}"
 inherit cmake
 
 # CXXFLAGS are needed to compile eigen 3.3.1 headers properly
-CXXFLAGS += "-Wno-deprecated-declarations -Wno-misleading-indentation -Wno-int-in-bool-context"
+CXXFLAGS += "-Wno-deprecated-declarations -Wno-misleading-indentation -Wno-int-in-bool-context -Wno-ignored-attributes"

